src/server/master_server/master_metadata_service_impl.cc

```diff
@@ -19,13 +26,90 @@ inline server::MetadataManager* MasterMetadataServiceImpl::metadata_manager() {
   return server::MetadataManager::GetInstance();
 }
 
+inline server::ChunkServerManager& MasterMetadataServiceImpl
+    ::chunk_server_manager() {
+  return server::ChunkServerManager::GetInstance();
+}
+
 grpc::Status MasterMetadataServiceImpl::HandleFileCreation(
     const protos::grpc::OpenFileRequest* request,
     protos::grpc::OpenFileReply* reply) {
+  // Step 1. Create file metadata
   const std::string& filename(request->filename());
   google::protobuf::util::Status status(
       metadata_manager()->CreateFileMetadata(filename));
-  return common::utils::ConvertProtobufStatusToGrpcStatus(status);
+  if (!status.ok()) {
+    return common::utils::ConvertProtobufStatusToGrpcStatus(status);
+  }
+
+  // Step 2. Create the first file chunk for this file
+  google::protobuf::util::StatusOr<std::string> chunk_or(
+      metadata_manager()->CreateChunkHandle(filename,0));
+  if (!chunk_or.ok()) {
+    return common::utils::ConvertProtobufStatusToGrpcStatus(status);
+  }
+  const std::string& chunk(chunk_or.ValueOrDie());
+
+  // Step 3. Allocate chunk servers
+  const ushort num_of_chunk_replica(3);
+  // TODO(someone): make this number configurable via config.yml. Probably not
+  // a priority right now
+  auto chunk_server_locations(
```

Comment:
nit: this can be empty, if no location is available. We should log and return fail for this.

Reply:
Yes, I think for file creation it may not qualify an error because the client may call openFileRequest again the next time, but yes let's keep this in mind.